src/main/resources/com/google/api/codegen/py/main.snip

```diff
@@ -235,12 +235,17 @@
                         @end
                     @end
 
+                    if metadata is None:
+                        metadata = []
+                    metadata = list(metadata)
                     @if apiMethod.headerRequestParams
-
-                        routing_header = google.api_core.gapic_v1.routing_header.to_grpc_metadata(
-                            [{@routingHeader(apiMethod.headerRequestParams)}],
-                        )
-                        metadata.append(routing_header)
+                        try:
+                            routing_header = google.api_core.gapic_v1.routing_header.to_grpc_metadata(
```

Comment:
It's not entirely clear that the error is supposed to catch just the case where the nested params are missing. Move more of the logic to an `else` clause. Something like:
```python
try:
    routing_header = [('name', name), ('book.read', book.read)]
except AttributeError:
    pass
else:
    routing_metadata = google.api_core.gapic_v1.routing_header.to_grpc_metadata(routing_header)
    metadata.append(routing_metadata)
```

Reply:
definitely